platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/DecryptMigrationManagerImpl.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.configuration.migration;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.BufferedOutputStream;
+import java.io.Closeable;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.Validate;
+import org.codice.ddf.migration.MigrationException;
+import org.codice.ddf.migration.MigrationOperation;
+import org.codice.ddf.migration.MigrationReport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** The decrypt migration manager decrypts an exported file. */
+public class DecryptMigrationManagerImpl implements Closeable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DecryptMigrationManagerImpl.class);
+
+  private final MigrationReport report;
+
+  private final MigrationZipFile zip;
+
+  private final ZipOutputStream zipOutputStream;
+
+  private final Path exportFile;
+
+  private final Path decryptFile;
+
+  private boolean close = false;
+
+  @SuppressWarnings(
+      "squid:S2095" /* up to the caller of this method to close it; exception is thrown if the stream cannot be created in the first place */)
+  private static ZipOutputStream newZipOutputStreamFor(Path decryptFile) {
+    Validate.notNull(decryptFile, "invalid null decrypt file");
+    try {
+      return new ZipOutputStream(
+          new BufferedOutputStream(new FileOutputStream(decryptFile.toFile())));
+    } catch (SecurityException | FileNotFoundException e) {
+      throw new MigrationException(Messages.DECRYPT_FILE_CREATE_ERROR, decryptFile, e);
```

Comment:
❓ Should we really treat the `FileNotFound` exception as a migration problem, or should it be more like a user-input problem, i.e., `IllegalArgumentException`? Either way, we should make sure that the message reported is clear. Right now, I'm not sure an admin would know that things failed because they provided a bad file name.

Reply:
Migration exceptions will be reported to the admin on the console. Runtime exceptions thrown by the service are to be dealt as programmatic errors.

Right now the admin cannot enter a filename so this is all a moot point but let's say he/she could. I would prefer reporting back to the admin that we could not create a file because the name is invalid or because there is already a read-only file in the directory. This will be done automatically by recording these errors in the report (or throwing MigrationException).